artemis-core-client/src/main/java/org/apache/activemq/artemis/core/settings/impl/DiskFullMessagePolicy.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.settings.impl;
+
+public enum DiskFullMessagePolicy {
```

Comment:
@iliya-gr I added PageFullMessagePolicy as part of [ARTEMIS-3178](https://issues.apache.org/jira/browse/ARTEMIS-3178).

I wonder if we need a separate option between DiskFull and PageFull. The only difference here I see is that pageFull is on a hard limit... But I am thinking, for the sake of easy of usage.. we should just use the same option here, and not add a new option... (keep everything you did, but just don't add a new configuration option).

I mean.. if disk is full, I would consider Page as as being full, and we should either drop of fail based on that already existing option.

Reply:
I agree. That was my first thought about it. I will rewrite it for page full policy

Reply:
And full page  default is FAIL, while default disk full behavior is BLOCK.

Reply:
@iliya-gr I will mark this as a draft.. can you make it ready to merge when you finish it?

Reply:
@jbertram  when I added the max-page-limit, I was already stretching the definitions...


because you had the page-limit = block...

But then they wanted to have a page limit after paged.

You have pagePolicy = BLOCK | PAGE | FAIL

I added pageLIMIT = BLOCL | FAIL


and now we want to a diskFull limit at BLOCK | PAGE | FULL


The combination of these 3 options gets really complex.. multipley 3X3 and you have a lot of possible combinations.


I say, lets keep it simple!

Reply:
I agree that additional option is not a very good solution. But switching to the page full policy will break backward capability. Now if paging is on producer will always be blocked, switching to page full policy will change behavior to FAIL by default. And there are no BLOCK page full policy currently.

Reply:
@iliya-gr a question actually... there is disk-full done before... how you get to a situation where you enter the page method.. if you had disk-full option on the broker blocking it?

Reply:
You won't if you previously blocked it. But I don't fully understand your question. You also may not block it before.